README.md

```diff
@@ -90,6 +90,63 @@ found 1725 well-typed programs.
+
+We will walk you through the first use. The other is very similar. On line 172 you will see the following expression. The expression generate 10000 random expressions (`e+`) and checks whether for all expression `e+`, if the expression compiles and terminates within certain amoount of reduction steps, the result must be of the same type as the expression. The nonterminate `e+` is defined in language `SP-conjecture`, which inherite the definition from language `SP` ([line 29 of `grammar.rkt`](https://github.com/brownplt/insta-model/blob/c9f21f5479b2dd4f9ddbfabacba88d22b3cb1811/grammar.rkt#L29)
+
+```racket
+(redex-check SP-conjecture
+             e+
+             (term (compile-implies-terminate-implies-well-typed-e e+))
+             #:attempts 10000)
+```
+
+In the repository we tested 10000 random expressions and 15000 random programs. The numbers are different in the paper (1600 and 11000) but you can always change the numbers.
+
+> (page 10; Section 4) we translated 265 tests from the Static Python regression suite to the syntax of the model and confirmed that the results do match, which suggests that the model conforms to actual Static Python.
+
+You can see the number (265) if you run 
+
+```bash
+make statistics
+```
+
+> (page 10; Section 4) For most of the 265 tests, the translation is automatic. A few tests required hand- pruning to remove features that the model does not handle (52 total)
+
+You can see the number (52) with 
+
+```bash
+ls ./conformance_suite/edited_test_* | wc -l
+```
+
+> (page 10; Section 4) Static Python has 537 other tests (802 total) that we did not use because they fall outside the scope of the model.
+
+`make statistics` also produces this number (802).
+
+> (page 13; Section 4.2.1) Most [casts] call for a tag check, i.e., a Python isinstance test. The sole exception is optional types, which require a tag check and a test for the none value.
+
+If you look at [the definition of `compile-check` in `./compile.rkt`](https://github.com/brownplt/insta-model/blob/c9f21f5479b2dd4f9ddbfabacba88d22b3cb1811/compile.rkt#L1225), the `dynamic` case is trivial. The function case is unreachable because there is no function type in the surface syntax. That's what SP does! (We introduced function types internally to make checking function/method calls easier.) The remaining two cases are more interesting. When the target type is a class, `compile-check` calls [`check-exactness`](https://github.com/brownplt/insta-model/blob/c9f21f5479b2dd4f9ddbfabacba88d22b3cb1811/compile.rkt#L1244), which effectively performs a tag-check. The remaining case is for `Optional[T]`, which has a test for the none value and a tag check. Because the `T` in `Optional[T]` must be a class, the check must be a tag check.
```

Comment:
"We introduced ...." makes it sound like SP doesn't have function types internally. But it does!
(I'll fix)